datashader/bundling.py

```diff
@@ -0,0 +1,266 @@
+    return replaced_edges
+
+
+@nb.jit
+def smooth_segment(segments, tension):
+    seg_length = len(segments) - 2
+    for i in range(1, seg_length):
+        previous, current, next_point = segments[i - 1], segments[i], segments[i + 1]
+        current[0] = ((1 - tension) * current[0]) + (tension * (previous[0] + next_point[0]) / 2)
+        current[1] = ((1 - tension) * current[1]) + (tension * (previous[1] + next_point[1]) / 2)
+
+
+@nb.jit
+def smooth(edge_segments, tension):
+    for segments in edge_segments:
+        smooth_segment(segments, tension)
+
+
+@ngjit
+def advect_segments(segments, vert, horiz):
+    for i in range(1, len(segments) - 1):
+        x = int(segments[i][0] * ACCURACY)
+        y = int(segments[i][1] * ACCURACY)
+        segments[i][0] = segments[i][0] + horiz[x, y] / ACCURACY
+        segments[i][1] = segments[i][1] + vert[x, y] / ACCURACY
+        segments[i][0] = max(0, min(segments[i][0], 1))
+        segments[i][1] = max(0, min(segments[i][1], 1))
+
+
+def advect_and_resample(vert, horiz, segments, iterations=50):
+    for it in range(iterations):
+        advect_segments(segments, vert, horiz)
+        if it % 2 == 0:
+            segments = resample_edge(segments)
+    return segments
+
+
+@delayed
+def advect_resample_all(gradients, edge_segments):
+    vert, horiz = gradients
+    return [advect_and_resample(vert, horiz, edges) for edges in edge_segments]
+
+
+def batches(l, n):
+    """Yield successive n-sized batches from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+@delayed
+def draw_to_surface(edge_segments, bandwidth):
+    img = np.zeros((ACCURACY + 1, ACCURACY + 1))
+    for segments in edge_segments:
+        for point in segments:
+            img[int(point[0] * ACCURACY), int(point[1] * ACCURACY)] += 1
+    return gaussian(img, sigma=bandwidth / 2)
+
+
+@delayed
+def get_gradients(img):
+    img /= np.max(img)
+
+    horiz = sobel_h(img)
+    vert = sobel_v(img)
+
+    magnitude = np.sqrt(horiz**2 + vert**2) + 1e-5
+    vert /= magnitude
+    horiz /= magnitude
+    return (vert, horiz)
+
+
+def _from_pandas(nodes, edges):
+    def minmax_scale(series):
+        minimum, maximum = np.min(series), np.max(series)
+        return (series - minimum) / (maximum - minimum)
+
+    nodes['x'] = minmax_scale(nodes['x'])
+    nodes['y'] = minmax_scale(nodes['y'])
+
+    df = pd.merge(nodes, edges, left_index=True, right_on=['source'])
+    df = df.rename(columns={'x': 'src_x', 'y': 'src_y'})
+
+    df = pd.merge(nodes, df, left_index=True, right_on=['target'])
+    df = df.rename(columns={'x': 'dst_x', 'y': 'dst_y'})
+
+    df = df.filter(items=['src_x', 'src_y', 'dst_x', 'dst_y'])
+
+    edge_segments = []
+    for edge in df.get_values():
+        segments = [[edge[0], edge[1]], [edge[2], edge[3]]]
+        edge_segments.append(np.array(segments))
+    return edge_segments
+
+
+def _to_pandas(edge_segments):
```

Comment:
I renamed the two methods to `_convert_graph_to_edge_segments` and `_convert_edge_segments_to_dataframe`.